tests/backup_helper.go

```diff
@@ -5914,6 +5915,166 @@ func RestartAllVMsInNamespace(namespace string, waitForCompletion bool) error {
+	}
+	log.Infof("VM Name - %s", vm.Name)
+	ipAddress := vmInstance.Status.Interfaces[0].IP
+	log.Infof("IP Address - %s", ipAddress)
+
+	// Getting username of the VM
+	// Username has to be added as a label to the VM Spec
+	username := vmInstance.Labels["username"]
+	log.Infof("Username - %s", username)
+
+	// Get password of the VM
+	// Password has to be added as a value in the ConfigMap named kubevirt-creds whose key the name of the VM
+	cm, err := core.Instance().GetConfigMap("kubevirt-creds", "default")
+	if err != nil {
+		return err
+	}
+	password := cm.Data[vm.Name]
+
+	// SSH command to be executed
+	sshCmd := fmt.Sprintf("sshpass -p '%s' ssh -o StrictHostKeyChecking=no %s@%s %s", password, username, ipAddress, cmd)
+	log.Infof("SSH Command - %s", sshCmd)
+
+	// If the cluster provider is openshift then we are creating ssh pod and running the command in it
+	if os.Getenv("CLUSTER_PROVIDER") == "openshift" {
+		log.Infof("Cluster is openshift hence creating the SSH Pod")
+		err = initSSHPod()
+		if err != nil {
+			return err
+		}
+		testCmdArgs := []string{"sshpass", "-p", password, "ssh", "-o", "StrictHostKeyChecking=no", fmt.Sprintf("cirros@%s", ipAddress), "hostname"}
+		cmdArgs := []string{"sshpass", "-p", password, "ssh", "-o", "StrictHostKeyChecking=no", fmt.Sprintf("cirros@%s", ipAddress), cmd}
+
+		// To check if the ssh server is up and running
+		t := func() (interface{}, bool, error) {
+			output, err := k8sCore.RunCommandInPod(testCmdArgs, sshPodName, "ssh-container", "default")
+			if err != nil {
+				log.Infof("Error encountered")
+				if isConnectionError(err.Error()) {
+					log.Infof("Test connection output - \n%s", output)
+					return "", true, err
+				} else {
+					return "", false, err
+				}
+			}
+			log.Infof("Test connection success output - \n%s", output)
+			return "", false, nil
+		}
+		_, err = task.DoRetryWithTimeout(t, 10*time.Minute, 30*time.Second)
+		if err != nil {
+			return err
+		}
+
+		// Executing the actual command
+		output, err := k8sCore.RunCommandInPod(cmdArgs, sshPodName, "ssh-container", "default")
+		if err != nil {
+			return err
+		}
+		log.Infof("Output of cmd %s - \n%s", cmd, output)
+	} else {
+		workerNode := node.GetWorkerNodes()[0]
+		t := func() (interface{}, bool, error) {
+			output, err := runCmdGetOutput(sshCmd, workerNode)
+			if err != nil {
+				log.Infof("Error encountered")
+				if isConnectionError(err.Error()) {
+					log.Infof("Output of cmd %s - \n%s", cmd, output)
+					return "", true, err
+				} else {
+					return "", false, err
+				}
+			}
+			log.Infof("Output of cmd %s - \n%s", cmd, output)
+			return "", false, nil
+		}
+		_, err = task.DoRetryWithTimeout(t, 10*time.Minute, 30*time.Second)
+		return err
+	}
+	return nil
+}
+
+// initSSHPod creates a pod with ssh server installed and running along with sshpass utility
+func initSSHPod() error {
+	var p *corev1.Pod
+	var err error
+	if p, err = k8sCore.GetPodByName(sshPodName, "default"); p == nil {
```

Comment:
Let's not run it in default ns but either let's run this pod in same namespace as VM we want to ssh to or we can create a different namespace for this pod to run in altogether

Reply:
The reason I did this was to avoid spawning multiple pods in multiple namespaces. Now if I have to run ssh commands in 10 VMs which are in 10 different namespaces, then 1 SSH pod in default namespace would suffice. As opposed to created 10 SSH pods.
But yes, I can switch from default to some other dedicated namespace.

Reply:
Creating a ns called `ssh-pod-namespace` and bring up the pod there.